src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxPopup.java

```diff
@@ -185,8 +185,14 @@ protected Rectangle adjustPopupAndGetBounds() {
             updateContents(true);
         }
 
+        int popupBoundsY = comboBox.getBounds().height;
+        if (comboBox.isEditable() && comboBox.getBorder() != null) {
```

Comment:
Does it not affect a non-editable combo box with with a border that has large insets?

Reply:
In testing, it does also affect this case. In the normal use cases in testing, this editable Aqua comboBox with a TitledBorder seemed to be the only problematic one, but with custom large borders, this seems to also be an issue.

I recently discussed with @lawrence-andrew and learned that in the setBorder doc for JComponent, UI components are not fully supported with borders. It is instead recommended to place the component in a JPanel. I think this is one of those cases, but the fix in this PR fixes the found issue for the case of an editable Aqua comboBox with a border.

If you could give insights on this Lawrence, that'd be appreciated as well.

Reply:
Actually I see the same behavior (ComboBox's selected item overlapped with popup menu) for non-editable ComboBox with or without your fix(whether it has a border or not).
We can probably fix it along with this issue.


[JDK_8302558.java.txt](https://github.com/openjdk/jdk/files/10962183/JDK_8302558.java.txt)

Reply:
I see this now. I just tried to apply the same fix like with editable comboBoxes, but the positioning of the popup doesn't seem to change. Looking into why

Reply:
There is a property for comboboxes that determines whether a combobox should have a popup display "on" or "below" the component. I found this in another open test that explicitly sets this property to TRUE or FALSE to test positioning. The current behavior is correct since a non-editable combobox's popup appears over the combobox rather than under. Prasanta gave an example native component in another comment in this PR.

Reply:
@azvegint In reference to the combobox's button, the email I got back from Apple says "in Aqua, there is no concept of a non editable Combo Box... what you are depicting as a JComboBox is actually a popup/pulldown". However, "NSComboBox, because of its direct affinity to NSTextField, has the square bezel corners" rather than "the softer rounded corners that the button family uses". My understanding from this is, since JComboBox relies on TextField, it uses the native TextField's square corners.

Reply:
I guess now in JDK with your testcase, popup button are shown opposite to what native components shows ie editable combobox shows non-editable pop-up button and non-editable combobox shows editable pull-down button

 
![image](https://user-images.githubusercontent.com/43534309/225811385-0aea444e-80b3-4608-8a19-31abddaa4501.png)

whereas in native for non-editable one we have pop-up
![image](https://user-images.githubusercontent.com/43534309/225811657-62c3f8c5-9ed2-4ad9-b291-48b55569c79b.png)
and for editable we have pull-down
![image](https://user-images.githubusercontent.com/43534309/225812101-9939bd87-d3ef-4f1c-ac32-c5bd37ac647e.png)

Reply:
This is because the test is set to test for isPopDown set to true, which changes the button appearance for non-editable to be a single down arrow. With this set to false, the button would match native. I have updated the test to test the native instead. Thanks.

For the editable comboboxes, the only button that is supported is the up and down arrow one shown. So, this one cannot be changed.

They now show as: 
<img width="199" alt="Screen Shot 2023-03-17 at 11 23 11 AM" src="https://user-images.githubusercontent.com/96267980/225987450-0ebd0c80-cbea-4601-88d8-3155616c3c0e.png">

Reply:
The above Apple statement only talks about non-editable combobox but I guess, we have issue, as you told, with editable combobox buttons which as per this code relies on BUTTON_COMBO_BOX state, so I guess we should ask Apple about BUTTON_POP_DOWN support for editable one or any other alternative via JRS

https://github.com/openjdk/jdk/blob/c09f83ec25749af349fb5609e3641b5bb6d34072/src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxButton.java#L144-L157

Reply:
Sure, I can setup a question for clarification. Looking at the original set of questions to Apple, the dropdown button symbol was included, but wasn't answered. Must have been lost in the list of questions at the time.

Reply:
ok